main.go

```diff
@@ -21,22 +21,43 @@ func (i *executionsArray) Set(value string) error {
 	return nil
 }
 
-var executions executionsArray
 var stopOnFailure bool
+var executions executionsArray
 
 func main() {
+	initFlags()
+
+	executions, err := parseExecuions()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+	stopOnFailure := parseStopOnFailure()
+
+	exitcode := runAllCommands(executions, stopOnFailure)
+
+	os.Exit(exitcode)
+}
+
+func initFlags() {
 	flag.Var(&executions, "execute", "eg. -execute='/usr/local/bin/mybin test123 -o -a' -execute='ls -l'")
 	flag.BoolVar(&stopOnFailure, "stop-on-failure", false, "If true, multiple execute steps get combined with && otherwise ||")
+}
+
+func parseExecuions() (executionsArray, error) {
 	flag.Parse()
 
 	if len(executions) == 0 {
-		fmt.Print("No commands found")
-		os.Exit(1)
+		return nil, fmt.Errorf("No commands found")
 	}
 
-	exitcode := runAllCommands(executions, stopOnFailure)
+	return executions, nil
+}
 
-	os.Exit(exitcode)
+func parseStopOnFailure() bool {
+	flag.Parse()
+
+	return stopOnFailure
```

Comment:
not sure if this method is needed. `flag.Parse` is already executed in `parseExecuions` so stopOnFailure should already be set